bubble_sort.rb

```diff
@@ -0,0 +1,43 @@
+def bubble_sort(arr)
+  index = 0
+  switched = false
+  while index < arr.length
+    return arr if index + 1 == arr.length
+
+    if arr[index] > arr[index + 1]
+      arr.insert(index, arr[index + 1])
+      arr.delete_at(index + 2)
+      switched = true
+    end
+    index += 1
+    if switched
+      index = 0
+      switched = false
+    end
+  end
+  arr
+end
+
+def bubble_sort_by(arr)
+  raise 'ERROR' unless block_given?
+
+  index = 0
+  switched = false
+  while index < arr.length
+    return arr if index + 1 == arr.length
+
+    if yield(arr[index], arr[index + 1]).positive?
+      arr.insert(index, arr[index + 1])
+      arr.delete_at(index + 2)
+      switched = true
+    end
+    index += 1
+    if switched
+      index = 0
+      switched = false
+    end
+  end
+  arr
```

Comment:
- [ ] it should raise an error if no block is given

<img width="460" alt="1" src="https://user-images.githubusercontent.com/58504655/103696330-fd2f1080-4fbf-11eb-89d6-ea3e397c67f9.PNG">